web/web.go

```diff
@@ -240,9 +241,16 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
+	hostname, err := getServerHostname()
 	if err != nil {
 		panic(err)
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
 }
+
+func getServerHostname() (string, error) {
+	if *hostname != "" {
+		return *hostname, nil
+	}
+	return os.Hostname()
+}
```

Comment:
Take a look at https://github.com/fatih/vim-go - it seems to be the most popular one-stop suite of Go functionality for vim at the moment.